l2geth/rollup/l2_gas_pricer.go

```diff
@@ -0,0 +1,43 @@
+package rollup
+
+import (
+	"math"
+)
+
+type GetTargetGasPerSecond func() float64
+
+type L2GasPricer struct {
+	curPrice                 float64
+	floorPrice               float64
+	getTargetGasPerSecond    GetTargetGasPerSecond
+	maxPercentChangePerEpoch float64
+}
+
+func NewGasPricer(curPrice float64, floorPrice float64, getTargetGasPerSecond GetTargetGasPerSecond, maxPercentChangePerEpoch float64) *L2GasPricer {
+	return &L2GasPricer{
+		curPrice:                 math.Max(curPrice, floorPrice),
+		floorPrice:               floorPrice,
+		getTargetGasPerSecond:    getTargetGasPerSecond,
+		maxPercentChangePerEpoch: maxPercentChangePerEpoch,
+	}
+}
+
+// Calculate the next gas price given some average gas per second over the last epoch
```

Comment:
If you could write out the formulas here in a comment, that would be super helpful for code review